docs/howto/add_confirm_dialog_in_a_form.md

```diff
@@ -0,0 +1,119 @@
+class `ez-js-open-modal` and a `value` attribute containing a selector matching
+it. As soon as the user clicks on an element with the class `ez-js-close-modal`
+the modal is closed.
+
+## Example
+
+### HTML code
+
+Let's take the following form:
+
+
+```html
+<form action="/whatever/action" method="post">
+    <!-- some markup -->
+
+    <button type="submit" name="DangerousOperation">
+        Dangerous operation
+    </button>
+</form>
+```
+
+To show a confirm dialog before doing the *Dangerous operation*, you have to:
+
+1. Replace the *DangerousOperation* `submit` button by a button with:
+    * the `type` attribute set to `button` so that clicking on it does not
+      submit the form
+    * the `class` attribute containing `ez-js-open-modal` so the app recognizes
+      this button as a button to open a dialog.
+    * the `value` attribute set with a selector matching the `<dialog>` element
+      to show
+1. Add the actual `<dialog>` element in the form. It should contain the
+   original submit button with the class `ez-js-close-modal` so that clicking on
+   it submits the form and closes the dialog.
+
+This gives the following HTML code:
+
+```html
+<form action="/whatever/action" method="post">
+    <!-- some markup -->
+
+    <button type="button" value="#confirm-dangerous-operation" class="ez-js-open-modal">
+        Dangerous operation
+    </button>
+
+    <!-- it can be before or after the button -->
+    <dialog class="ez-modal" id="confirm-dangerous-operation">
+        <!--
+            it's also possible to have header here:
+            <header>Sure?</header>
+            it could also contain a button with the `ez-js-close-modal` class
+            to close the dialog in addition/instead of the Cancel button below
+        -->
+        <section>
+            <p>Are you sure you want to do the dangerous operation?</p>
+        </section>
+        <footer>
+            <button type="button" class="ez-js-close-modal">Cancel</button>
+            <button type="submit" name="DangerousOperation" class="ez-js-close-modal">
+                Dangerous operation
+            </button>
+        </footer>
+    </dialog>
+</form>
+```
+
+### Symfony form and Twig template
+
+In a Symfony application, it's common to use [the Form Component to integrate
+forms](https://symfony.com/doc/current/forms.html) and most forms in the
+application are build this way. The Hybrid Platform UI application is also
+generated with some Twig templates. In that context, the previous example could
+be generated with the following template code:
+
+```twig
+{{ form_start(myForm, {'action': path('form_handling_route')}) }}
+    <!-- some markup -->
+
+    {{ form_widget(myForm.delete, {'label': 'Dangerous Operation'}) }}
+{{ form_end(myForm) }}
+```
+
+The changes explained in the previous section can be applied directly in the
+Twig template but to ease the addition of a confirm dialog in such context, the
```

Comment:
I'd end the sentence after "Twig template", then start a new one:

"[...] can be applied directly in the Twig template. To ease the addition of a confirm dialog in such context, the"

("to ease the addition" sounds a bit "heavy" to me. You could maybe just remove it:

"[...] "can be applied directly in the Twig template. The Hybrid UI application provides [...]"

But @ezsystems/documentation-team probably has better insight than me on this :)